corehq/apps/hqadmin/management/commands/check_case_index_ids.py

```diff
@@ -29,14 +39,64 @@ def handle(self, *args, **options):
         cases_with_invalid_references = (
             domain_query
             .term('indices.referenced_id', invalid_referenced_ids)
-            .source(['_id', 'type', 'indices', 'owner_id', 'opened_by'])
+            .source(['_id', 'type', 'indices', 'owner_id', 'opened_by', 'xform_ids'])
             .run().hits
         )
 
         with open(options['filename'], 'w') as csvfile:
             writer = csv.writer(csvfile)
+            headers = [
+                'case id',
+                'case type',
+                'creating form id',
+                'referenced id',
+                'referenced_type',
+                'index relationship',
+                'index identifier',
+                'owner id',
+                'owner name',
+                'opened by id',
+                'opened by name',
+            ]
+            if debug:
+                headers.append('app version')
+            writer.writerow(headers)
 
             for case in cases_with_invalid_references:
                 for index in case['indices']:
                     if index['referenced_id'] in invalid_referenced_ids:
-                        writer.writerow([case['_id'], case['type'], index, case['owner_id'], case['opened_by']])
+                        form_id = case['xform_ids'][0]
+                        row = [
+                            case['_id'],
+                            case['type'],
+                            form_id,
+                            index['referenced_id'],
+                            index['referenced_type'],
+                            index['relationship'],
+                            index['identifier'],
+                            case['owner_id'],
+                            cached_owner_id_to_display(case['owner_id']),
+                            case['opened_by'],
+                            cached_owner_id_to_display(case['opened_by']),
+                        ]
+                        if debug:
+                            form = FormAccessors(domain=domain).get_form(form_id)
+                            app_version_info = get_app_version_info(
+                                domain,
+                                form.build_id,
+                                form.form_data['@version'],
+                                form.metadata,
+                            )
+                            row.append(app_version_info.build_version)
+                        writer.writerow(row)
+
+        if close_all:
```

Comment:
(in full transparency i didn't get a chance to test this logic - just in case you run into any issues with it)